Lombiq.Hosting.MediaTheme.Deployer/Program.cs

```diff
@@ -49,102 +82,158 @@ private static void HandleParseError(IEnumerable<Error> errors)
+        var newDirectoryPath = CreateNewDirectoryPath(options);
 
         try
         {
-            // Determine whether the directory exists.
-            if (Directory.Exists(newDirectoryPath))
-            {
-                WriteLine("That directory already exists.");
-                return;
-            }
+            if (Directory.Exists(newDirectoryPath)) Directory.Delete(newDirectoryPath, recursive: true);
 
-            // Try to create the directory.
             Directory.CreateDirectory(newDirectoryPath);
 
-            WriteLine("The directory was created successfully. {0}", newDirectoryPath);
+            WriteLine("The \"{0}\" directory was created successfully.", newDirectoryPath);
         }
-        catch (Exception exception)
+        catch (Exception)
         {
-            WriteLine("The directory creation failed: {0}", exception.ToString());
-            return;
+            WriteLine("Creating the directory {0} failed.", newDirectoryPath);
+            throw;
+        }
+
+        var themePath = options.ThemePath;
+
+        if (string.IsNullOrEmpty(themePath))
+        {
+            throw new ArgumentException("The theme's path must be provided.");
         }
 
-        var pathToTheme = values.PathOfTheTheme;
+        var recipeSteps = new JArray();
+
+        // Creating Feature step to enable the Media Theme theme and Bridge module.
+        dynamic featureStep = new JObject();
+        featureStep.name = "Feature";
+        featureStep.enable = new JArray("Lombiq.Hosting.MediaTheme.Bridge", "Lombiq.Hosting.MediaTheme");
+        recipeSteps.Add(featureStep);
+
+        // Creating Themes step to set Media Theme as the site theme.
+        dynamic themesStep = new JObject();
+        themesStep.name = "Themes";
+        themesStep.Site = "Lombiq.Hosting.MediaTheme";
+        recipeSteps.Add(themesStep);
 
         // Creating media theme step.
         dynamic mediaThemeStep = new JObject();
         mediaThemeStep.name = "mediatheme";
-        mediaThemeStep.BaseThemeId = values.BaseThemeId;
-        mediaThemeStep.ClearMediaThemeFolder = values.ClearMediaHostingFolder;
+        mediaThemeStep.ClearMediaThemeFolder = options.ClearMediaHostingFolder;
+        recipeSteps.Add(mediaThemeStep);
+
+        var baseThemeId = string.IsNullOrEmpty(options.BaseThemeId) ? null : options.BaseThemeId;
+
+        if (baseThemeId == null)
+        {
+            var manifestPath = Path.Combine(themePath, "Manifest.cs");
+            var manifestContent = await File.ReadAllTextAsync(manifestPath);
+            var basteThemeMatch = Regex.Match(
+                manifestContent, @"BaseTheme\s*=\s*""(?<baseThemeId>.*)""", RegexOptions.ExplicitCapture, TimeSpan.FromSeconds(1));
+
+            if (basteThemeMatch.Success)
+            {
+                baseThemeId = basteThemeMatch.Groups["baseThemeId"].Value;
+            }
+        }
+
+        mediaThemeStep.BaseThemeId = baseThemeId;
```

Comment:
![image](https://user-images.githubusercontent.com/4007293/210286574-9e8c1eb8-4c0a-4ac5-9f6f-56a755eb1ed0.png)

I guess if the `if (basteThemeMatch.Success)` is false then this should throw an exception?

Reply:
No, Media Themes are allowed to have no base theme. Then `baseThemeId` is purposefully `null`.